src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPoolMetrics.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+namespace System.Buffers;
+
+internal sealed class PinnedBlockMemoryPoolMetrics
+{
+    // Note: Dot separated instead of dash.
+    public const string MeterName = "System.Buffers.PinnedBlockMemoryPool";
```

Comment:
I'm not sure if these metrics should be on a new meter. Couldn't they be on a meter the same name as Kestrel's meter?

A new meter name means folks would need to configure OTEL to export the kestrel meter, and this `System.Buffers.PinnedBlockMemoryPool` meter. Having memory stats related to kestrel be on the Kestrel meter makes more sense to me.

Reply:
I see there are multiple frameworks that use `System.Buffers.PinnedBlockMemoryPool`. Maybe it does make sense for it to be its own meter. Although if you have multiple frameworks using the pool at the same time then numbers will get merged together. Is that would you intend?

A solution would be to have a tag that says who used the memory. That way you can view memory usage of Kestrel HTTP layer, vs sockets layer, vs IIS (if running together in the same process)

Reply:
I think the meter should be renamed to indicate that it's specific to ASP.NET Core. Maybe `Microsoft.AspNetCore.MemoryPool`? That is simple and obvious what the meter is for.

I don't think people care that the underlying pool is implemented using pinned blocks.

Reply:
Changed the name. But can we limit metric feedback to the issue https://github.com/dotnet/aspnetcore/issues/61594?

The owner tag seems interesting, but how would it actually work? For example, the app layer can grab the memory pool factory via DI and get a pool instance, so if you wanted a different name for app code then the interface would need to allow setting a name. If this is something you think we should pursue, can you comment on the issue I linked?